src/libraries/System.Threading.RateLimiting/tests/FixedWindowRateLimiterTests.cs

```diff
@@ -523,6 +523,41 @@ public override async Task CanCancelAcquireAsyncAfterQueuing()
             Assert.Equal(1, limiter.GetStatistics().CurrentAvailablePermits);
         }
 
+        [Fact]
+        public override async Task CanFillQueueWithOldestFirstAfterCancelingFirstQueuedRequestManually()
+        {
+            var limiter = new FixedWindowRateLimiter(new FixedWindowRateLimiterOptions
+            {
+                PermitLimit = 3,
+                QueueProcessingOrder = QueueProcessingOrder.OldestFirst,
+                QueueLimit = 3,
+                Window = TimeSpan.FromMilliseconds(2),
+                AutoReplenishment = false
+            });
+
+            var lease = limiter.AttemptAcquire(1);
+            Assert.True(lease.IsAcquired);
+
+            lease = limiter.AttemptAcquire(2);
+            Assert.True(lease.IsAcquired);
+
+            var cts = new CancellationTokenSource();
+            var wait = limiter.AcquireAsync(2, cts.Token);
+            cts.Cancel();
+
+            var ex = await Assert.ThrowsAsync<TaskCanceledException>(() => wait.AsTask());
+            Assert.Equal(cts.Token, ex.CancellationToken);
+
+            var wait2 = limiter.AcquireAsync(1);
+            Replenish(limiter, 2L);
+
+            lease = await wait2;
```

Comment:
I noticed that FixedWindow actually doesn't suffer from this issue, or at least I couldn't see a way it could be affected. I still think we should keep the change in FixedWindow for consistency, just wanted to point it out.

Reply:
Yes, you are right; I added the needed change for FixedWindow.